ecosystem/typescript/sdk/src/transaction_builder/builder.ts

```diff
@@ -292,3 +295,98 @@ export class TransactionBuilderABI {
     throw new Error("Invalid ABI.");
   }
 }
+
+export type RemoteABIBuilderConfig = Partial<Omit<ABIBuilderConfig, "sender">> & {
+  sender: MaybeHexString | AccountAddress;
+};
+
+interface AptosClientInterface {
+  getAccountModules: (arg0: string) => Promise<Gen.MoveModuleBytecode[]>;
+  getAccount: (accountAddress: MaybeHexString) => Promise<Gen.AccountData>;
+  getChainId: () => Promise<number>;
+}
+
+/**
+ * This transaction builder downloads JSON ABIs from the fullnodes.
+ * It then translates the JSON ABIs to the format that is accepted by TransactionBuilderABI
+ */
+export class TransactionBuilderRemoteABI {
+  // We don't want the builder to depend on the actual AptosClient. There might be circular dependencies.
+  constructor(
+    private readonly aptosClient: AptosClientInterface,
+    private readonly builderConfig: RemoteABIBuilderConfig,
+  ) {}
+
+  // Cache for 10 minutes
```

Comment:
Is there a way to force it to evict the cache?

Reply:
yes. people can call the `clear` method. But we need to tag this. people can call clear('tag')